packages/flutter/lib/src/material/filled_button.dart

```diff
@@ -0,0 +1,754 @@
+  _FilledButtonDefaultOverlay(this.overlay);
+
+  final Color overlay;
+
+  @override
+  Color? resolve(Set<MaterialState> states) {
+    if (states.contains(MaterialState.hovered)) {
+      return overlay.withOpacity(0.08);
+    }
+    if (states.contains(MaterialState.focused) || states.contains(MaterialState.pressed)) {
+      return overlay.withOpacity(0.24);
+    }
+    return null;
+  }
+}
+
+@immutable
+class _FilledButtonDefaultElevation extends MaterialStateProperty<double> with Diagnosticable {
+  _FilledButtonDefaultElevation(this.elevation);
+
+  final double elevation;
+
+  @override
+  double resolve(Set<MaterialState> states) {
+    if (states.contains(MaterialState.disabled)) {
+      return 0;
+    }
+    if (states.contains(MaterialState.hovered)) {
+      return elevation + 2;
+    }
+    if (states.contains(MaterialState.focused)) {
+      return elevation + 2;
+    }
+    if (states.contains(MaterialState.pressed)) {
+      return elevation + 6;
+    }
+    return elevation;
+  }
+}
+
+@immutable
+class _FilledButtonDefaultMouseCursor extends MaterialStateProperty<MouseCursor?> with Diagnosticable {
+  _FilledButtonDefaultMouseCursor(this.enabledCursor, this.disabledCursor);
+
+  final MouseCursor? enabledCursor;
+  final MouseCursor? disabledCursor;
+
+  @override
+  MouseCursor? resolve(Set<MaterialState> states) {
+    if (states.contains(MaterialState.disabled)) {
+      return disabledCursor;
+    }
+    return enabledCursor;
+  }
+}
+
+class _FilledButtonWithIcon extends FilledButton {
+  _FilledButtonWithIcon({
+    super.key,
+    required super.onPressed,
+    super.onLongPress,
+    super.onHover,
+    super.onFocusChange,
+    super.style,
+    super.focusNode,
+    bool? autofocus,
+    Clip? clipBehavior,
+    required Widget icon,
+    required Widget label,
+  }) : assert(icon != null),
+       assert(label != null),
+       super(
+         autofocus: autofocus ?? false,
+         clipBehavior: clipBehavior ?? Clip.none,
+         child: _FilledButtonWithIconChild(icon: icon, label: label)
+      );
+
+  _FilledButtonWithIcon.tonal({
+    super.key,
+    required super.onPressed,
+    super.onLongPress,
+    super.onHover,
+    super.onFocusChange,
+    super.style,
+    super.focusNode,
+    bool? autofocus,
+    Clip? clipBehavior,
+    required Widget icon,
+    required Widget label,
+  }) : assert(icon != null),
+       assert(label != null),
+       super.tonal(
+         autofocus: autofocus ?? false,
+         clipBehavior: clipBehavior ?? Clip.none,
+         child: _FilledButtonWithIconChild(icon: icon, label: label)
+       );
+
+  @override
+  ButtonStyle defaultStyleOf(BuildContext context) {
+    final EdgeInsetsGeometry scaledPadding = ButtonStyleButton.scaledPadding(
+      const EdgeInsetsDirectional.fromSTEB(12, 0, 16, 0),
+      const EdgeInsets.symmetric(horizontal: 8),
+      const EdgeInsetsDirectional.fromSTEB(8, 0, 4, 0),
```

Comment:
where do these values come from?

Reply:
Copied from the other ButtonStyle classes 😄? That said, I don't think these padding values are in the tokens yet. When they are we will have to do a sweep across the widgets to include the layout data. Currently these are hard coded.